Track device rotation (Droid)

jasells opened the pull request:
Fixes #50 on Android by tracking _physical_ orientation of the device, not the window/app-layout default orientation.

Depends/based on #63.

janusw reviewed (changes requested):
Please keep your PRs independent of each other. Since the custom clean targets from the first commit are the subject of #65, please keep them out of this one.

janusw commented:
> > Please keep your PRs independent of each other. Since the custom clean targets from the first commit are the subject of #65, please keep them out of this one.
> 
> Thought at least a version of that was merged.

Well, no, I rejected the custom clean target completely, since I was not fully happy with it (sorry if I was not clear about that in #63).

> I hate Github

Hm, if you don't like GitHub's way of presenting things, I'd recommend a viewer like gitk or similar, which shows you the full history and all branches. The advantage of an 'offline' viewer is also that you can inspect your local tree _before_ pushing to GitHub. I use gitk regularly, but there are various alternatives as well.

Btw, your latest rebase squashed everything into one commit, apparently. To be honest, I liked the previous approach better, where you had several smaller commits (and separate the library changes from the test app changes) 🙈 🤷

jasells commented:
> Btw, your latest rebase squashed everything into one commit, apparently. To be honest, I liked the previous approach better, where you had several smaller commits (and separate the library changes from the test app changes

Yah, that was easier than cherry picking the various commits, i just reverted the ones u disliked and then squashed.  HIstory is much easier to read, IMHO.   I removed the rotation-locking from the test app.

I'd like to get this done sooner rather than later, I have other things I need to work.  If this continues to drag out,  I may just move ahead with my own fork, and let you merge (or not) from it as you like.

janusw commented:
> > Btw, your latest rebase squashed everything into one commit, apparently. To be honest, I liked the previous approach better, where you had several smaller commits (and separate the library changes from the test app changes
> 
> Yah, that was easier than cherry picking the various commits, i just reverted the ones u disliked and then squashed. History is much easier to read, IMHO.

I totally disagree on this point. Your previous branch was much easier to understand and follow, because things that were unrelated and independent of each other were well-separated in single commits.

History may become hard to read when interleaved merges of various branches are involved. Here we have a simple linear chain of a few commits (which are supposed to be finalized by a single merge-commit to master, which will also show that they belong together). That's a perfect example of readable history.


> I'd like to get this done sooner rather than later, I have other things I need to work. If this continues to drag out, I may just move ahead with my own fork, and let you merge (or not) from it as you like.

Well, trust me, I also have other things to work on 🤣. I maintain this open-source project fully in my spare time. And in fact I have been pretty fast in reviewing and commenting all your recent PRs. But actually I don't have the obligation to do any of this. And I certainly cannot be pressured into doing anything by threats of doing whatever ...

jasells commented:
> I maintain this open-source project fully in my spare time. And in fact I have been pretty fast in reviewing and commenting all your recent PRs. But actually I don't have the obligation to do any of this. And I certainly cannot be pressured into doing anything by threats of doing whatever ...

No threats my friend.   Simply suspecting what you just confirmed: that we simply have differing priorities.  I do respect and appreciate the work you put in to port this code from XForms -> Maui.  I just worked full app-stack port as well, so I know the pain, and like you wish the OP was around to maintain this code instead of you or me. 😅

At the same time, I have code soon to go into full production that has very ugly, complicated work-around code to address this issue, and it seems as if my time may be best spend just setting up my own fork to publish this fix, and possibly others if your time or inclinations are not available.  Nothing personal in the slightest.

janusw commented:
Looks like you prefer to create yet another fork of this already-forked repo 🤯. Not a very clever idea, if you ask me. Anyway, it's your choice. Have fun ...

jasells commented:
> The packages are named/versioned via `git describe`. That's not too far from what you envision. The next step might be to publish these packages to a package server, e.g. [GitHub's nuget registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry), which is probably easiest to integrate with GitHub Actions, and would represent a separate "pre-release" registry, as opposed to the stable releases on nuget.org.

Just some notes from my own experience:

1. Nuget natively supports release/prerelease (or CI) notions via "-StringTag" appended to the version.  Publishing both to the same place is much more intuitive for consumers.
2. Github's package management is crude at best and in my experience is private, not public.  Management can only be done via cli, so if there happens to be a bad build that should be unlisted, it is much harder to do.
3. Integrating anything into Github actions is not that easy, but in relative terms integrating the public Nuget gallery is no harder or easier than Github's solution, with far better management options.
